idaes/generic_models/unit_models/tests/test_skeleton_unit_model.py

```diff
@@ -0,0 +1,305 @@
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Test for skeleton unit model.
+"""
+
+import pytest
+from pyomo.environ import ConcreteModel, Constraint, Var, Set, value, \
+    SolverStatus, TerminationCondition
+from idaes.core import FlowsheetBlock
+from idaes.generic_models.unit_models import SkeletonUnitModel
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util import get_solver
+from idaes.core.util.exceptions import ConfigurationError
+
+__author__ = "Jaffer Ghouse"
+
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+# -----------------------------------------------------------------------------
+
+
+class TestSkeletonDefault(object):
+    """
+    Test for SkeletonUnitModel without callback for initialization
+
+    """
+    @pytest.fixture(scope="class")
+    def skeleton_default(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+
+        m.fs.skeleton = SkeletonUnitModel(default={"dynamic": False,
+                                           "custom_initializer": False})
+        m.fs.skeleton.comp_list = Set(initialize=["c1", "c2"])
+
+        # input vars for skeleton
+        m.fs.skeleton.flow_comp_in = \
+            Var(m.fs.time, m.fs.skeleton.comp_list, initialize=1.0)
+        m.fs.skeleton.temperature_in = Var(m.fs.time, initialize=298.15)
+        m.fs.skeleton.pressure_in = Var(m.fs.time, initialize=101)
+
+        # output vars for skeleton
+        m.fs.skeleton.flow_comp_out = \
+            Var(m.fs.time, m.fs.skeleton.comp_list, initialize=1.0)
+        m.fs.skeleton.temperature_out = Var(m.fs.time, initialize=298.15)
+        m.fs.skeleton.pressure_out = Var(m.fs.time, initialize=101)
+
+        # Surrogate model equations
+        # Flow equation
+
+        def rule_flow(m, t, i):
+            return m.flow_comp_out[t, i] == m.flow_comp_in[t, i]
+
+        m.fs.skeleton.eq_flow = \
+            Constraint(m.fs.time, m.fs.skeleton.comp_list, rule=rule_flow)
+
+        m.fs.skeleton.eq_temperature = Constraint(
+            expr=m.fs.skeleton.temperature_out[0] ==
+            m.fs.skeleton.temperature_in[0] + 10)
+        m.fs.skeleton.eq_pressure = Constraint(
+            expr=m.fs.skeleton.pressure_out[0] ==
+            m.fs.skeleton.pressure_in[0] - 10)
+
+        inlet_dict = {"flow_mol_comp": m.fs.skeleton.flow_comp_in,
+                      "temperature": m.fs.skeleton.temperature_in,
+                      "pressure": m.fs.skeleton.pressure_in}
+        outlet_dict = {"flow_mol_comp": m.fs.skeleton.flow_comp_out,
+                       "temperature": m.fs.skeleton.temperature_out,
+                       "pressure": m.fs.skeleton.pressure_out}
+
+        m.fs.skeleton.add_ports(name="inlet", member_dict=inlet_dict)
```

Comment:
Just from the view point of unit testing methods, I would suggest only calling these in the `test_ports` method. That way the method is being called as part of the test for its effects.